### ISSUE TITLE
Cannot store promise-like values

### ISSUE THREAD

**Issue by annevk** [WICG/kv-storage#49]
I'm not sure to what extent this is considered a problem, but unlike with Indexed DB, it seems you cannot store a value like `{ then: "hi", now: "boo" }` due to promises unwrapping all the things.

At the very least this might be worth calling out?

**Comment by domenic** [comment]
Yeah, not much we can do here, but definitely worth calling out, and also writing some tests for.

**Comment by domenic** [comment]
Actually, I realized this is fine. The OP example works with no problems, since "hi" is not a function. And, trying to store an object with a then method would not work anyway since functions are not structured serializable.

I guess we could still add tests, but at this point I don't think there's much worth adding a note to the spec for. What do you think?

**Comment by annevk** [comment]
Ah yeah, in particular, if IsCallable returns true for something, it'll throw.

**Comment by annevk** [comment]
(I guess I didn't realize IsCallable is a requirement for thenables.)
